sympy/core/numbers.py

```diff
@@ -38,6 +38,10 @@
 from sympy.external.pythonmpq import PythonMPQ as MPQ
```

Comment:
I _think_ that if you import `MPQ` from gmpy `instead` it will basically do the lines added in this commit.

https://github.com/sympy/sympy/blob/dcdf59bbc6b13ddbc329431adf72fcee294b6389/sympy/external/gmpy.py#L83-L104